vendor/github.com/containers/image/v5/docker/docker_transport.go

```diff
@@ -152,6 +182,9 @@ func (ref dockerReference) NewImageDestination(ctx context.Context, sys *types.S
 
 // DeleteImage deletes the named image from the registry, if supported.
 func (ref dockerReference) DeleteImage(ctx context.Context, sys *types.SystemContext) error {
+	if ref.isUnknownDigest {
```

Comment:
Similarly to `NewImageSource`, I think pushing the check to the start of `deleteImage` itself would make the requirement closer to the code which relies on it.